File: commands/misc/flip.js
module.exports = {
	name: 'coinflip',
	description: 'Flips a coin for heads or tails',
	aliases: ['flip', 'heads', 'tails'],
	usage: '++coinflip',
	example: '++coinflip or ++flip',
	inHelp: 'yes',
	execute(message, args) {
    function doRand() {
      const rand = ['HEADS!', 'TAILS!'];
      return rand[Math.floor(Math.random()*rand.length)];
    }
    const embed = {
      color: '#C977BB',
      title: 'You got...',
      description: doRand(),
      timestamp: new Date()
    };
    message.channel.send({ embeds: [embed] })
  },
};

File: lib/src/commands/misc/flip.dart
import 'dart:math';

import 'package:nyxx/nyxx.dart';
import 'package:nyxx_commands/nyxx_commands.dart';

final _random = Random();

final coinFlip = ChatCommand(
  'coin-flip',
  'Flip a coin',
  aliases: ['flip', 'coinflip', 'heads', 'tails'],
  id('coin-flip', (IChatContext context) async {
    const choices = ['HEADS!', 'TAILS!'];

    final choice = choices[_random.nextInt(2)];

    final embed = EmbedBuilder()
      ..color = DiscordColor.hotPink
      ..title = 'You got...'
      ..description = choice
      ..timestamp = DateTime.now();

    await context.respond(MessageBuilder.embed(embed));
  }),
);
